laalaa/apps/advisers/tasks.py

```diff
@@ -1,51 +1,44 @@
 # -*- coding: utf-8 -*-
 import csv
 import logging
-import os
+import time
 import tempfile
-from threading import Thread, Event
-from time import sleep
-
+import itertools
+from celery.task import TaskSet
+import re
+from django.utils.text import slugify
+import os
+from celery import Task
 from django.conf import settings
 from django.contrib.gis.geos import Point
-from django.db import connection, transaction
-from django.utils.text import slugify
-import xlrd
+from django.core.cache import cache
+from django.db import connection
 
 from . import models
 from . import geocoder
+import xlrd
 
 
 logging.basicConfig(filename='adviser_import.log', level=logging.WARNING)
 
 
-def cached(fn):
-    cache = {}
-
-    def wrapped(name):
-        if name not in cache:
-            cache[name] = fn(name)
-        return cache[name]
-
-    wrapped.cache = cache
-    wrapped.clear_cache = cache.clear
-    return wrapped
+def to_key(postcode):
+    return re.sub('[^0-9A-Z]+', '', postcode)
 
 
-@cached
 def geocode(postcode):
     point = None
-    try:
-        loc = models.Location.objects.filter(postcode=postcode)
-        if len(loc) and loc[0].point:
-            point = loc[0].point
+    loc = models.Location.objects.filter(postcode=postcode)
+    if len(loc) and loc[0].point:
+        point = loc[0].point
```

Comment:
I know I wrote this originally, but this could be cleaned up:
```python
loc = models.Location.objects.filter(postcode=postcode, point__isnull=False)
if len(loc):
    point = loc[0].point
```

And if `postcode` is unique on `Location`, this could be even simpler:
```python
try:
    point = models.Location.objects.get(postcode=postcode, point__isnull=False).point
    
except Location.DoesNotExist:
    ...
```